src/hotspot/share/runtime/sharedRuntime.cpp

```diff
@@ -233,6 +233,7 @@ const juint  float_infinity   = 0x7F800000;
 const julong double_sign_mask = CONST64(0x7FFFFFFFFFFFFFFF);
 const julong double_infinity  = CONST64(0x7FF0000000000000);
 
+#ifndef X86
```

Comment:
I wonder if the WIN64 workaround is actually needed/valid for non-X86 windows?

Reply:
It is true the comment says:
> 64-bit Windows on amd64 returns the wrong values for infinity operands.
I have left the workaround really just for AMD64 Windows now. I am going to get it regression tested on aarch64 if that is enough.

Reply:
So the _WIN64 workaround will be always there.